CONTRIBUTING.md

```diff
@@ -0,0 +1,124 @@
+
+If you are planning on contributing code to Everest, it is recommended to create a [fork](https://guides.github.com/activities/forking/) of the main repo to develop on.
+Once you have a fork, clone it to your machine and follow the instructions [here](https://github.com/EverestAPI/Everest#compiling-everest-yourself) to build and install it.
+
+
+# Types of Contributions
+These are guidelines for some of the different types of contributions you can make to Everest.
+
+## Bug Reports
+Make sure bug reports have enough information to be reproduced easily!  
+That means:
+- **Steps to reproduce.**
+- Description of the bug (expected behaviour vs actual behaviour).
+- `log.txt` file from the session where the bug was encountered (located in your Celeste install folder, with previous sessions stored in the `LogHistory` subfolder).
+
+## Ideas or Feature Requests
+When submitting ideas or feature requests, consider carefully whether the feature would be more suited for a code mod.  
+Features for Everest should match at least one of the following criteria:
+- Adds features for code mods.
+- Specific to the Everest project.
+- Not possible to achieve through a code mod.
+
+## Pull Requests
+When making a pull request, please note the following guidelines:
+- **Branch name:** use a descriptive name for your branch, following the format of `action-target-details` where possible.
+- **Description:** explain what your pull request changes and _why_.
+- **Credits:** if your modifications are in response to a request or issue, link to it or otherwise provide a "paper trail".
+- **Testing:** if you are submitting a bugfix, describe the issue thouroughly and provide recreation steps, *especially* if it is not an existing [issue](https://github.com/EverestAPI/Everest/issues).
+- **Response:** when you submit a pull request, you are responsible for making sure any questions and change requests on it are answered in a timely manner.
+
+
+# Making Changes
+These are explanations and guidelines for how to make changes to the Everest source code.
+
+## Adding Content
+Content added *must not* change the functionality of the base game, or change any public facing API.  
+While many mods also make use of non-public code it is not as important to maintain backwards compatibility for it.
+
+New features should only be added to improve or un-hardcode vanilla features, especially for general use in code mods (f.e. [StrawberryRegistry](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Mod/Registry/StrawberryRegistry.cs), [CustomNPC](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Mod/Entities/CustomNPC.cs), [Custom Events](https://github.com/EverestAPI/Everest/blob/dev/Celeste.Mod.mm/Patches/EventTrigger.cs)).  
+Gameplay mechanics, entities, and other features not present in the base game are usually better suited for a mod.
```

Comment:
What about quality of life improvements (f.e. the return to map hint that later inspired the vanilla implementation, or other QoL changes to Everest itself)? Or features which require a deep integration with the game's code, and thus would be better suited as part of the mod loader, but don't necessarily count as "improving vanilla features" (f.e. the map parenting system)? Maybe it's worth mentioning that exceptions will be made on a case by case basis.